wrappers/dlib/face/validate_face.cpp

```diff
@@ -0,0 +1,162 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
+
+#include "validate_face.h"
+#include "markup_68.h"
+
+
+/*
+    Calculates the average depth for a range of two-dimentional points in face, such that:
+        point(n) = face.part(n)
+    and puts the result in *p_average_depth.
+
+    Points for which no depth is available (is 0) are ignored and not factored into the average.
+
+    Returns true if an average is available (at least one point has depth); false otherwise.
+*/
+bool find_depth_from(
+    rs2::depth_frame const & frame,
+    float const depth_scale,
+    dlib::full_object_detection const & face,
+    int markup_from, int markup_to,
+    float * p_average_depth
+)
+{
+    uint16_t const * data = reinterpret_cast<uint16_t const *>(frame.get_data());
+
+    float average_depth = 0;
+    size_t n_points = 0;
+    for( int i = markup_from; i <= markup_to; ++i )
+    {
+        auto pt = face.part( i );
+        auto depth_in_pixels = *(data + pt.y() * frame.get_width() + pt.x());
+        if( !depth_in_pixels )
+            continue;
+        average_depth += depth_in_pixels * depth_scale;
+        ++n_points;
+    }
+    if( ! n_points )
+        return false;
+    if( p_average_depth )
+        *p_average_depth = average_depth / n_points;
+    return true;
+}
+
+
+// Forward declarations for the two implementations
+bool validate_face_5 ( rs2::depth_frame const & frame, float const depth_scale, dlib::full_object_detection const & face );
```

Comment:
Add a one-line description for each method

Reply:
Added